### PR TITLE
paraview hdf5 output

### PR DESCRIPTION
**Pull request by teseoch**

### CODE REVIEW

File: src/polyfem/io/OutData.cpp [hunk 1]
```diff
@@ -1228,7 +1184,12 @@ namespace polyfem::io
 			}
 		}
 
-		VTUWriter writer;
+		std::shared_ptr<paraviewo::ParaviewWriter> tmpw;
+		if (opts.use_hdf5)
+			tmpw = std::make_shared<paraviewo::HDF5VTUWriter>();
+		else
+			tmpw = std::make_shared<paraviewo::VTUWriter>();
+		paraviewo::ParaviewWriter &writer = *tmpw;
```
Comment (zfergus) [inline src/polyfem/io/OutData.cpp:1192]: Are you sure tmpw wont be garbage collected leaving writer invalid?

File: src/polyfem/io/OutData.cpp [hunk 2]
```diff
@@ -956,6 +957,9 @@ namespace polyfem::io
 
 		reorder_output = args["output"]["data"]["advanced"]["reorder_nodes"];
 
+		// TODO
+		use_hdf5 = false;
```
Comment (zfergus) [inline src/polyfem/io/OutData.cpp:961]: Finish this before merging?

File: src/polyfem/io/OutData.cpp (hunk 1 above)
Comment (zfergus) [reply src/polyfem/io/OutData.cpp:1192]: Is writer just a local variable?

**Review by zfergus — APPROVED** [review]

File: src/polyfem/io/OutData.cpp (hunk 1 above)
Comment (teseoch) [reply src/polyfem/io/OutData.cpp:1192]: it is, but I want it polymorphic. Is there a better way?

File: src/polyfem/io/OutData.cpp (hunk 2 above)
Comment (teseoch) [reply src/polyfem/io/OutData.cpp:961]: Yes, it requires changing the spec, will do once I have time

File: src/polyfem/io/OutData.cpp (hunk 1 above)
Comment (zfergus) [reply src/polyfem/io/OutData.cpp:1192]: Just use the shared  pointer?

**Pull request merged**